src/App.tsx

```diff
@@ -1,19 +1,171 @@
 import React from 'react';
-import './App.css';
+import classNames from 'classnames';
+import './App.scss';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todos } from './components/types/Todos';
 
 import users from './api/users';
+import todos from './api/todos';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+const preparedTodos = todos.map(item => {
+  return {
+    ...item,
+    user: users.find(person => item.userId === person.id) || null,
+  };
+});
+
+type State = {
+  name: string;
+  title: string;
+  addedTodos: Todos[],
+  titleError: boolean;
+  nameError: boolean;
+  maxTitleLength: number;
 };
 
-export default App;
+export class App extends React.Component<{}, State> {
+  state = {
+    title: '',
+    name: '',
+    addedTodos: [...preparedTodos],
+    titleError: false,
+    nameError: false,
+    maxTitleLength: 20,
+  };
+
+  addTodo = () => {
+    if (!this.state.title.trim().length) {
+      this.setState(state => ({
+        titleError: !state.titleError,
+      }));
+
+      return;
+    }
+
+    if (this.state.title.match(/[^a-zA-Zа-яА-Я0-9]/)) {
+      this.setState(state => ({
+        titleError: !state.titleError,
+      }));
+    }
+
+    if (!this.state.name.length) {
+      this.setState(state => ({
+        nameError: !state.nameError,
+      }));
+
+      return;
+    }
+
+    const todoId = () => {
+      if (this.state.addedTodos.length === 0) {
+        return 1;
+      }
+
+      const array = this.state.addedTodos.map(item => item.id);
+      const maxNum = Math.max(...array);
+
+      return maxNum + 1;
+    };
+
+    const addedUser = users.find(user => user.name === this.state.name);
+    const newTodo = {
+      userId: addedUser?.id,
+      id: todoId(),
+      title: this.state.title.trim().slice(0, this.state.maxTitleLength),
+      completed: false,
+      user: addedUser || null,
+    };
+
+    this.setState(state => ({
+      addedTodos: [...state.addedTodos, newTodo],
+      title: '',
+      name: '',
+    }));
+  };
+
+  removeTodo = (id: number) => {
+    this.setState(state => ({
+      addedTodos: [...state.addedTodos].filter(item => item.id !== id),
```

Comment:
filter doesn't mutate array

```suggestion
      addedTodos: state.addedTodos.filter(item => item.id !== id),
```

Reply:
fixed